gobblin-data-management/src/main/java/gobblin/data/management/copy/hive/HiveCopyEntityHelper.java

```diff
@@ -314,15 +317,26 @@
           this.existingTargetTable = Optional.absent();
         }
 
+        // Constructing CommitStep object for table registration
         Path targetPath = getTargetLocation(dataset.fs, this.targetFs, dataset.table.getDataLocation(),
             Optional.<Partition> absent());
         this.targetTable = getTargetTable(this.dataset.table, targetPath);
         HiveSpec tableHiveSpec = new SimpleHiveSpec.Builder<>(targetPath)
             .withTable(HiveMetaStoreUtils.getHiveTable(this.targetTable.getTTable())).build();
-        CommitStep tableRegistrationStep = new HiveRegisterStep(this.targetURI, tableHiveSpec, this.hiveRegProps);
-
+        CommitStep tableRegistrationStep = new HiveRegisterStep(this.targetURI, tableHiveSpec, this.hiveRegProps, false);
         this.tableRegistrationStep = Optional.of(tableRegistrationStep);
 
+        // Constructing CommitStep object for table alteration, not necessary to use.
+        if (this.existingTargetTable.isPresent()) {
+          HiveSpec alteredTableHiveSpec = new SimpleHiveSpec.Builder<>(targetPath)
```

Comment:
It is possible that attributes other than location need to be changed.
I think the fix is a bit simpler than this. If you provide a new table to register, `HiveRegister` will automatically alter the table if it already exists. All that you need to do is, if user specifies `UPDATE_TABLE`, then don't delete the existing table.